src/System.Text.Json/src/System/Text/Json/JsonConstants.cs

```diff
@@ -30,11 +30,33 @@ internal static class JsonConstants
         // Used to search for the end of a number
         public static ReadOnlySpan<byte> Delimiters => new byte[] { ListSeperator, CloseBrace, CloseBracket, Space, LineFeed, CarriageReturn, Tab, Slash };
 
-        public static ReadOnlySpan<byte> WhiteSpace => new byte[] { Space, LineFeed, CarriageReturn, Tab };
-
-        public static ReadOnlySpan<byte> EndOfComment => new byte[] { Asterisk, Slash };
-
         // Explicitly skipping ReverseSolidus since that is handled separately
         public static ReadOnlySpan<byte> EscapableChars => new byte[] { Quote, (byte)'n', (byte)'r', (byte)'t', Slash, (byte)'u', (byte)'b', (byte)'f' };
+
+        public const int RemoveFlagsBitMask = 0x7FFFFFFF;
+        public const int MaxWriterDepth = 1_000;
+        public const int MaxTokenSize = 2_000_000_000 / 6;  // 357_913_941 bytes
+        public const int MaxCharacterTokenSize = 2_000_000_000 / 6; // 357_913_941 characters
+
+        public const int MaximumInt64Length = 20;   // 19 + sign (i.e. -9223372036854775808)
+        public const int MaximumUInt64Length = 20;  // i.e. 18446744073709551615
+        public const int MaximumDoubleLength = 128;  // default (i.e. 'G'), using 128 (rather than say 32) to be future-proof.
```

Comment:
@tannergooding, I set these higher than necessary (i.e. 128) as per our discussion. Please review.

Reply:
Is this used for parsing or formatting?

For formatting, you only need 17 digits to guarantee roundtripping (unless the user asks for more). For parsing, you may have to track up to 768 significant digits in order to return the correctly rounded result.

Reply:
> Is this used for parsing or formatting?

This is for formatting only. I will add a comment (and update the names).

What **should** be the max required length for formatting a double using the `Utf8Formatter` **by default**, i.e. when the user doesn't pass a `StandardFormat`? Should it be 17 (i.e. the smallest amount that guarantees round tripping) where the implementation of `Utf8Formatter` is fixed to behave accordingly (breaking change?)? The `Utf8JsonWriter`, by default, calls the default implementation of `Utf8Formatter` (whatever default it sets for different .NET types). I want to make sure the max value I am using here will work for that case. Should I therefore keep it at 128? If we can guarantee 17, then it would be great if I can reduce the `MaximumDoubleLength` down.
**OR**
Do you think the `Utf8JsonWriter` should be formatting the JSON text in a round-trippable manner by default, anyway, regardless of the default behavior of `Utf8JsonWriter`? Then I could set it to something like 17, but it would mean the writer becomes slower.